crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -57,7 +57,56 @@ def prepare
   api_version "2.0"
   def services
     if request.post?
-      head :not_implemented
+
+      # prepare the scripts for various actions in one chef run
+      service_object = CrowbarService.new(Rails.logger)
+      service_object.prepare_nodes_for_os_upgrade
```

Comment:
Well, there's no execution of code on nodes in this one, but usual applying of crowbar proposal. So may think about similar errors like from the `prepare` action.